### PR TITLE
Wdl workflow

### PR DESCRIPTION
**Pull request by adthrasher**
Add WDL implementation of the XenoCP workflow

### CODE REVIEW

**Comment by adthrasher** [comment]
@a-frantz - If you have a chance, I'd appreciate a review on this. After this is merged, I intend to add an optional step to RNA-Seq V2 (BAM and FastQ) to run XenoCP.

**Review by a-frantz — CHANGES_REQUESTED** [review]
You have an `ncpu` parameter for a lot of tasks that don't use the resulting `n_cores` variable. Besides that and the incorrect/missing meta fields, the WDL looks good.

Didn't look at the `xenocp` Dockerfile or the rest of this repo, but I'm assuming that's all working as well.

File: wdl/tools/xenocp.wdl [hunk 1]
```diff
@@ -343,11 +322,13 @@ task cleanse {
...
         author: "Andrew Thrasher"
         email: "andrew.thrasher@stjude.org"
-        description: "This WDL tool runs creates a list of contamination reads by comparing host alignments to original alignments." 
+        description: "This WDL tool unmaps reads that have better mappings to the contaminant genome." 
     }
 
     parameter_meta {
-        input_bam: "Input BAM file"
+        input_bam: "BAM file to cleanse of contaminant reads"
+        unmap_reads: "The list of read namess to unmap in the input BAM"
```
Comment (a-frantz) [inline wdl/tools/xenocp.wdl:330]:
```suggestion
        unmap_reads: "The list of read names to unmap in the input BAM"
```

File: wdl/tools/xenocp.wdl [hunk 2]
```diff
@@ -0,0 +1,428 @@
...
+    command <<<
+        set -euo pipefail
+
+        n_cores=~{ncpu}
+        if [ -n ~{parsed_detect_nproc} ]
+        then
+            n_cores=$(nproc)
+        fi
+
+        samtools view -H ~{input_bam} -@ ${n_cores}| grep "@SQ" | cut -f 2  | cut -f 2 -d':' > chroms.txt
```
Comment (mcrusch) [inline wdl/tools/xenocp.wdl:31]: We don't need to know number of CPUs or cores for this step, right?  It's just a samtools view -H.

Comment (adthrasher) [reply wdl/tools/xenocp.wdl:31]: We don't need to, but it can multithread, so I've enabled it.

**Review by mcrusch — APPROVED** [review]

**Review by a-frantz — APPROVED** [review]

**Pull request merged**